packages/core/src/render3/di.ts

```diff
@@ -43,13 +43,11 @@ export function inject<T>(token: Type<T>, flags?: InjectFlags): T {
         if (size !== 0) {
           size = size >> LNodeFlags.SIZE_SHIFT;
           const start = flags >> LNodeFlags.INDX_SHIFT;
-          const directives = node.view.directives;
-          if (directives) {
-            for (let i = start, ii = start + size; i < ii; i++) {
-              const def = directives[(i << 1) | 1];
-              if (def.diPublic && def.type == token) {
-                return directives[i << 1];
-              }
+          const ngStaticData = node.view.ngStaticData;
+          for (let i = start, ii = start + size; i < ii; i++) {
```

Comment:
`ii` -> `endIndex`

@mhevery I'm assuming this is an optimization to avoid recalculating the sum in each iteration? I'd be pretty surprised if integer addition makes any meaningful difference in the execution time. Are there any resources for context here?

If keeping, I'd add a comment like
```ts
// Store endIndex to avoid recalculating on each iteration
```

Reply:
I find it more readable this way. So its on optimization for humans, not for computer. 

FYI addition is pretty fast on modern CPUs' but not necessarily on phones. This is because t make addition fast you have to increase silicon area which than increases power consumption.

Reply:
I'd disagree that this (with the `ii`) is simpler than
```ts
for (let i = start; i < start + size; i++)
```
It's shorter, removes one (admittedly tiny) level of indirection, and you don't have to spend any brain cycles to note that `ii`is not used inside the loop body. Either way, I'd still argue that `ii` is a confusing name that would be better as `endIndex`.

As for performance, without data to back it up it still feels like a speculative premature optimization.